### PR TITLE
Will only attempt to populate the date object if the date looks sane

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php [hunk 1]
```diff
@@ -218,7 +219,12 @@ public function getDateOfBirth()
     public function setDateOfBirth($dateOfBirth)
     {
         if (!($dateOfBirth instanceof \DateTime)) {
-            $dateOfBirth = new \DateTime($dateOfBirth);
+            //We validate it against a "good" format
+            if (1 === preg_match(self::REGEX_DATE_FORMAT, trim($dateOfBirth))) {
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php]: `$dateOfBirth` is now an object, how can you `trim` it?

Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php]: Sorry thats in `red` :smile:

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php [hunk 2]
```diff
@@ -218,7 +219,12 @@ public function getDateOfBirth()
...
+                $dateOfBirth = new \DateTime($dateOfBirth);
+            } else {
+                $dateOfBirth = null;
```
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php]: Return `null`, should it not throw an **Exception** to bubble up?

Comment (brettminnie) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php]: That was the problem, it was throwing an unhandled exception, if we set it to null, it fails validation

Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php]: :+1:

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php (hunk 1 above)
Comment (brettminnie) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php]: l2read 
`if (!($dateOfBirth instanceof \DateTime)) {`
translation
`if dateOfBirth is not an instance of a DateTime object`

you're welcome :D

Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php]: I read the part in **red** ` $dateOfBirth = new \DateTime($dateOfBirth);` which is obviously no longer there :stuck_out_tongue:

Comment (brettminnie) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php]: Then say it's a Deletion, accuracy Mr Jaoude

**Pull request merged**